### PR TITLE
Support eslint v6

### PR DESCRIPTION
**Pull request by cartogram**
Brings us to eslint@6.x. Not having this update is causing some dependabot PRs to fail.

Update tests 'parser' requiring an absolute path rather than a package name. Details found in the [migration guide](https://eslint.org/docs/6.0.0/user-guide/migrating-to-6.0.0#rule-tester-parser). 

>In ESLint v6.0.0, RuleTester disallows parser property with a package name. To address: If you use parser property with package names in test cases, update it with require.resolve() function to resolve the package name to the absolute path to the package.

Without these changes to the tests, these tests expectedly failed due to the parser option not being an absolute path; with the change to the tests, they all pass.

### CODE REVIEW

File: tests/lib/rules/webpack/no-unnamed-dynamic-imports.js [hunk 1]
```diff
@@ -1,91 +1,84 @@
...
-      return System.import<{default: any}>(
-        /* webpackChunkName: "bar" */ './bar',
-      ).then(bar => bar);
-    }
-    `,
-    parser: 'typescript-eslint-parser',
-  },
-];
-
-const invalidExamples = [
```
Comment (cartogram) [inline tests/lib/rules/webpack/no-unnamed-dynamic-imports.js:63]: only changed for consistency with the rest of this repo: we do not ever store the examples in a variable before passing them into the ruleTester.

**Review by GoodForOneFare — APPROVED** [review]
👍 Adding this to CHANGELOG would be great.  

Scanning through the migration guide, maybe we should enable `no-async-promise-executor`, as it's now recommended?  That doesn't have to happen here, though.

File: lib/config/rules/possible-errors.js [hunk 2]
```diff
@@ -6,26 +6,15 @@ module.exports = {
...
-      imports: 'always-multiline',
-      exports: 'always-multiline',
-      functions: 'always-multiline',
-    },
-  ],
   // Disallow assignment in conditional expressions
   'no-cond-assign': 'error',
   // Disallow use of console
-  'no-console': 'error',
+  'no-console': 'off',
```
Comment (GoodForOneFare) [inline lib/config/rules/possible-errors.js:17]: I have a, "some men just want to watch the world burn" fascination for seeing how this plays out... _but_, if we're following the spirit of the migration guide:
- This should still be enabled by default
- It should be disabled for [the node config](https://github.com/Shopify/eslint-plugin-shopify/blob/master/lib/config/node.js)

Comment (cartogram) [reply lib/config/rules/possible-errors.js:17]: 😬 https://github.com/Shopify/eslint-plugin-shopify/pull/332

**Pull request merged**